### PR TITLE
management_categories_admin

### PR DESCRIPTION
**Pull request by Ayuminmin**
I created caegories_manegement.
I created category_edit_page and category_new_page.

### CODE REVIEW

File: app/controllers/admin/categories_controller.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
+class Admin::CategoriesController < AdminController
+  def index
+    @categories = Category.all.order(created_at: :desc)
+  end
+  def new
+    @category = Category.new
+  end
+  def create
+  @category = Category.new(category_params)
```
Comment (ivanraybuglosa) [inline app/controllers/admin/categories_controller.rb]: Insert an indention here

File: app/controllers/admin/categories_controller.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    if @category.save
+      flash[:success] = "Successfully Add Category"
+      redirect_to admin_categories_url
+    else
+      render 'new'
+    end
+  end
```
Comment (ivanraybuglosa) [inline app/controllers/admin/categories_controller.rb:18]: have an extra space after each method

File: app/views/admin/categories/_errors_messages.html.erb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<% if @category.errors.any? %>
+    <div class="alert alert-danger">
+        <% @category.errors.full_messages.each do |msg| %>
+            <li><%= msg %></li>
+        <% end %>
+    </div>
+<% end %>
+
```
Comment (ivanraybuglosa) [inline app/views/admin/categories/_errors_messages.html.erb]: Delete this file and use the _error_messages in the shared folder.

File: app/views/admin/categories/edit.html.erb [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+            <%= render 'errors_messages' %>
+            <div class="form-group">
+                <%= f.label :title %>
+                <%= f.text_field :title, class: 'form-control' %>
+            </div>
+            <div class="form-group">
+                <%= f.label :description %>
+                <%= f.text_area :description, class: 'form-control' %> 
+            </div>
+            <%= f.submit "Add category", class: 'btn btn-info btn-block' %>
```
Comment (ivanraybuglosa) [inline app/views/admin/categories/edit.html.erb]: Change text to "Update Category"

File: app/views/admin/categories/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+            <th>
+                <%= category.title %>
+            </th>
+            <th>
+                <%= category.description %> 
+            </th>
+            <th>
+                <%= link_to "Add Word", "#", class: 'btn btn-dark' %>
+                <%= link_to "Edit", edit_admin_category_path(category), class: 'btn btn-dark' %>
+                <%= link_to "Delet", admin_category_path(category),method: :delete, class: 'btn btn-dark' %>
```
Comment (ivanraybuglosa) [inline app/views/admin/categories/index.html.erb]: Change spelling to "Delete"

File: app/views/admin/users/index.html.erb [hunk 6]
```diff
@@ -22,6 +22,9 @@
     <%= will_paginate %>
 </div> 
 
+<<<<<<< Updated upstream
 
 
 
+=======
+>>>>>>> Stashed changes
```
Comment (ivanraybuglosa) [inline app/views/admin/users/index.html.erb]: Remove these spaces.

File: config/routes.rb [hunk 7]
```diff
@@ -7,6 +7,7 @@
 
   namespace :admin do
     resources :users, only: [:index,:update]
+    resources :categories, except: [:show]
```
Comment (ivanraybuglosa) [inline config/routes.rb]: Please use all routes for categories. You'll use show route in the future

**Comment by Ayuminmin** [comment]
I created space and used shared/errors_messages.

**Pull request merged**